### PR TITLE
chore: update godocs for legacy apis on controller module

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes: #2234 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by chatton — APPROVED** [review]
LGTM 🥇 thanks for fixing typo!

File: modules/apps/27-interchain-accounts/controller/keeper/account.go [hunk 1]
```diff
@@ -16,6 +16,10 @@ import (
 // - A new MsgChannelOpenInit is routed through the MsgServiceRouter, executing the OnOpenChanInit callback stack as configured.
 // - An error is returned if the port identifier is already in use. Gaining access to interchain accounts whose channels
 // have closed cannot be done with this function. A regular MsgChannelOpenInit must be used.
+//
+// NOTE: this is a legacy API that is only intended to function correctly in workflows where an underlying application has been set
+// Prior to to v6.x.x of ibc-go, the controller module was only functional as middleware, with authentication performed
+// by the underlying authentication application. For a full summary of the changes in v6.x.x, please see ADR009.
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/controller/keeper/account.go:22]: nit

```suggestion
// NOTE: this is a legacy API that is only intended to function correctly in workflows where an underlying application has been set.
// Prior to to v6.x.x of ibc-go, the controller module was only functional as middleware, with authentication performed
// by the underlying application. For a full summary of the changes in v6.x.x, please see ADR009.
```

File: modules/apps/27-interchain-accounts/controller/keeper/relay.go [hunk 2]
```diff
@@ -16,6 +16,10 @@ import (
 // If the base application has the capability to send on the provided portID. An appropriate
 // absolute timeoutTimestamp must be provided. If the packet is timed out, the channel will be closed.
 // In the case of channel closure, a new channel may be reopened to reconnect to the host chain.
+//
+// NOTE: this is a legacy API that is only intended to function correctly in workflows where an underlying application has been set
+// Prior to to v6.x.x of ibc-go, the controller module was only functional as middleware, with authentication performed
+// by the underlying authentication application. For a full summary of the changes in v6.x.x, please see ADR009.
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/controller/keeper/relay.go:22]: same nit

```suggestion
// NOTE: this is a legacy API that is only intended to function correctly in workflows where an underlying application has been set.
// Prior to to v6.x.x of ibc-go, the controller module was only functional as middleware, with authentication performed
// by the underlying application. For a full summary of the changes in v6.x.x, please see ADR009.
```

**Review by damiannolan — APPROVED** [review]
LGTM! 🚀

File: modules/apps/27-interchain-accounts/controller/keeper/account.go [hunk 3]
```diff
@@ -16,6 +16,10 @@ import (
 // - A new MsgChannelOpenInit is routed through the MsgServiceRouter, executing the OnOpenChanInit callback stack as configured.
 // - An error is returned if the port identifier is already in use. Gaining access to interchain accounts whose channels
 // have closed cannot be done with this function. A regular MsgChannelOpenInit must be used.
+//
+// NOTE: this is a legacy API that is only intended to function correctly in workflows where an underlying application has been set
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/keeper/account.go:20]:
```suggestion
// NOTE: this is a legacy API that is only intended to function correctly in workflows where an underlying application has been set. Calling this API will result in all packet callbacks being routed to the underlying application.
// Please use MsgRegisterInterchainAccount for use cases which do not need to route to an underlying application
```

**Review by damiannolan — CHANGES_REQUESTED** [review]
@charleenfei could you pull in the changes from #2448 here please? See https://github.com/cosmos/ibc-go/pull/2448#issuecomment-1263304151

Thanks!

**Review by damiannolan — APPROVED** [review]
🙏

**Review by colin-axner — APPROVED** [review]

**Pull request merged**